pkgs/tools/security/ghidra/build-extension.nix

```diff
@@ -23,12 +23,13 @@ let
         gradle
       ];
 
-      preBuild = ''
+      preBuild = args.preBuild or "" + ''
         # Set project name, otherwise defaults to directory name
         echo -e '\nrootProject.name = "${pname}"' >> settings.gradle
       '';
```

Comment:
```suggestion
      preBuild = ''
        # Set project name, otherwise defaults to directory name
        echo -e '\nrootProject.name = "${pname}"' >> settings.gradle
        ${args.preBuild or ""}
      '';
```

for example, if the `preBuild` needs to undo the `rootProject.name` changes, it's better to have user-defined `preBuild` after the default `preBuild`

Reply:
Nvm, this breaks my flow, as the settings.gradle file has to be created in the dir with the plugin code, but most projects don't have that at the top level.
https://github.com/vringar/ghidra-setup/blob/4e91cb063cfad510677b49ba5cf7efbfa7930ea5/extensions/lightkeeper/default.nix#L17-L19
https://github.com/vringar/ghidra-setup/blob/4e91cb063cfad510677b49ba5cf7efbfa7930ea5/extensions/ret-sync/default.nix#L19-L21